src/com/palmergames/bukkit/towny/utils/JailUtil.java

```diff
@@ -43,37 +44,33 @@ public class JailUtil {
 	 * @param reason JailReason resident is jailed for.
 	 * @param jailer CommandSender of who did the jailing or null.
 	 */
-	public static void jailResident(Resident resident, Jail jail, int cell, int hours, JailReason reason, CommandSender jailer) {
+	public static void jailResident(Resident resident, Jail jail, int cell, int hours, double bail, JailReason reason, CommandSender jailer) {

```

Comment:
The master branch method should be re-added, pointing to this method, with a bail amount supplied. Changing this method will break other plugins that use jails.

Reply:
I'm also not sure how you want to handle bail being 0. Most of the changes in this method assume bail will cost money and the server has economy enabled.

Reply:
New commit made to address this issue.
- Master branch method readded pointing to new method with bail amount supplied.
- Check if server has bail configuration option enabled and has economy enabled, send appropriate messaging.